### PR TITLE
Feature/storage/weekly tests run

### PR DESCRIPTION
**Pull request by amnguye**
Follow up to attempting to fix #19542 so that the endpoints are created correctly

### CODE REVIEW

File: sdk/storage/test-resources.json
Comment (amnguye) [inline sdk/storage/test-resources.json:384]: This was a roundabout way to grab the value from the URL that I wanted, if there's a not-as-complicated way, let me know.

**Review by benbp — APPROVED** [review]

File: sdk/storage/test-resources.json
Comment (kasobol-msft) [reply sdk/storage/test-resources.json:384]: Could you extract a function to encapsulate this logic and avoid repetition . see https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-user-defined-functions

File: sdk/storage/test-resources-post.ps1
Comment (kasobol-msft) [inline sdk/storage/test-resources-post.ps1:56]: How does this work if there's no zone ? wound't this produce `https://foo..blob.core.windows.com` ? I.e. empty segment between dots? On the other hand tests seem to be passing, so how does this work?

File: sdk/storage/test-resources-post.ps1
Comment (kasobol-msft) [inline sdk/storage/test-resources-post.ps1:93]: this looks different than other entires. i.e. theres no blob/queue/table inside.

**Review by kasobol-msft — COMMENTED** [review]
I have looked around https://dev.azure.com/azure-sdk/internal/_build/results?buildId=811450&view=results but didn't notice any account with zone in outputs from arm template execution. is this intentional?

File: sdk/storage/test-resources-post.ps1
Comment (amnguye) [reply sdk/storage/test-resources-post.ps1:93]: good catch. done.

File: sdk/storage/test-resources-post.ps1
Comment (kasobol-msft) [reply sdk/storage/test-resources-post.ps1:56]: Would be good to call it `zone with period` if you do this. so one doesn't have to analyze data/debug to discover this in the future.

**Review by seanmcc-msft — COMMENTED** [review]
Could we also run the live tests pipeline and make sure this change does not effect it?

File: sdk/storage/test-resources.json
Comment (amnguye) [inline sdk/storage/test-resources.json:23]: @benbp Since this endpoint suffix value `storage.azure.net` only applies to private preview subscriptions, is there a way to set this as we do with the other parameters?

**Comment by benbp** [comment]
@seanmcc-msft the live tests pipeline is in the passing checks, though I don't know if it's been ran for the latest/future iterations.

**Comment by amnguye** [comment]
> I have looked around https://dev.azure.com/azure-sdk/internal/_build/results?buildId=811450&view=results but didn't notice any account with zone in outputs from arm template execution. is this intentional?

The zone is not considered a part of the account name. So it is intentional. Looking from the run the zone is still being added to the URL when constructed in the test config file.

File: sdk/storage/test-resources.json
Comment (kasobol-msft) [inline sdk/storage/test-resources.json:386]: nit: I was execting that zone_with_period would have trailing "." and wondered if endpoints are right. but I can see here that "." is prefix. Not a big deal but might confuse some readers. Shouldn't it be called period_with_zone ?;-)

**Review by kasobol-msft — APPROVED** [review]
lgtm

File: sdk/storage/test-resources.json
Comment (amnguye) [reply sdk/storage/test-resources.json:386]: I was having an internal debate about that haha, so I flipped a coin and settled on the what I have up there.

Just so that everything makes sense, I will just change it so that its the period is a suffix and not a prefix that way it matches it's name. (Also I would rather call it zone with period, and not period with zone, just cause I think it sounds weird).

Comment (amnguye) [reply sdk/storage/test-resources.json:386]: Actually the more I think about this, I dont think this should even work for regular endpoints.

This would always grab the value after the storage account, so if it was a regular endpoint (e.g. `storageaccount.blob.core.windows.net`) then this value would end up being blob. so this would be wrong because when we go to reconstruct the uri then we get `storageaccount.blob.blob.core.windows.net`

I have to change this so we end up getting the endpoint suffix for each service instead.

Comment (kasobol-msft) [reply sdk/storage/test-resources.json:386]: if that's true then having test runs pass seems to be disturbing. Maybe we have lot of dead wood in this configuration. Might be worth spending some time to consolidate this config (remove the redundancy/unused xml tags).

Comment (kasobol-msft) [reply sdk/storage/test-resources.json:386]: I hope changes in this PR are actually picked up by these jobs.. maybe we should run a sanity check and do some temporary commit/change in the config that derails them all and see if jobs are failing.

Comment (amnguye) [reply sdk/storage/test-resources.json]: I made a github issue so I can get to working on the config after this PR is merged https://github.com/Azure/azure-sdk-for-net/issues/20128 .

I have a feeling that we are parsing out the storage account information, and then remaking the URLs / Connectionstrings in the tests, and that's why we're not catching some of these changes.

**Comment by amnguye** [comment]
Closing this PR so we dont merge to master. We still need to investigate https://github.com/Azure/azure-sdk-for-net/issues/20128

**Pull request closed without merging**
